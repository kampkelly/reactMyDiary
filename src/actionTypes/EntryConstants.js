export const NEW_ENTRY = 'NEW_ENTRY';
