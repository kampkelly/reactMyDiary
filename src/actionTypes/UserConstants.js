export const SIGNIN = 'SIGNIN';
export const SIGNUP = 'SIGNUP';
export const SIGNOUT = 'SIGNOUT';
export const PROFILE = 'PROFILE';
export const NOTIFICATION = 'NOTIFICATION';