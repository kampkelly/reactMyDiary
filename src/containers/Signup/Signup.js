import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './Signup.scss';
import '../../styles/form.scss';
import { SignupUser } from '../../requests/UserRequests';
import { validateForm } from '../../helpers/validateForm';


/**
 * @class Signup
 * @extends {Component}
 */
class Signup extends Component {
  /**
   *Creates an instance of Signin.
   * @memberof Signin
   */
  constructor() {
    super();
    this.state = {
      email: '',
      fullName: '',
      password: '',
      confirmPassword: '',
      dateOfBirth: '',
      success: false
    };
    this.submitSignup = this.submitSignup.bind(this);
    this.change = this.change.bind(this);
    this.goToHome = this.goToHome.bind(this);
  }

  /**
   * @description - This method runs whenever redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   * @memberof Header
   */
  static getDerivedStateFromProps(props, state) {
    return {
      failure: props.failure,
      success: props.success,
    };
  }

  goToHome() {
    this.props.history.push('/signup');
  }

  /**
   *
   *
   * @memberof Signup
   */
  submitSignup(e) {
    const errorMessage = 'One or more of the required fields are empty!';
    const values = { email: this.state.email, fullName: this.state.fullName, password: this.state.password, confirmPassword: this.state.confirmPassword };
    const noneEmpty = validateForm(errorMessage, values, e);
    if (noneEmpty === true) {
      // validation was successful
      const email = this.state.email.toLowerCase().replace(/\s+/g, '');
      const password = this.state.password.toLowerCase();
      const confirmPassword = this.state.confirmPassword.toLowerCase();
      this.props.SignupUser(email, password, confirmPassword, this.state.dateOfBirth, this.state.fullName);
    }
  }

  change(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  /**
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   * @memberof Signup
   */
  render() {
    const { email, password, confirmPassword, dateOfBirth, fullName } = this.state;
    return (
      <div>
        <section id="signup">
          <h1 className="white-text text-center">Welcome to MyDiary</h1>
          <h5 className="white-text text-center">Keep your personal thoughts safe...</h5>
          <h2 className="text-center white-text">Create An Account</h2>
            <div className="blue-forms form-cover text-center">
              <form className="signup_form" id="signup-form">
                <p className="danger-text form_error_text"><small></small></p>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                    <label className="white-text">Full Name:</label>
                  </div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <input type="text" required="true" className="control-form" id="fullname" name="fullName" value={fullName} onChange={this.change} autoComplete="fullName" />
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                    <label className="white-text">Email Address</label>
                  </div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <input type="email" required="true" className="control-form" id="email" name="email" value={email} onChange={this.change} autoComplete="email" />
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                    <label className="white-text">Date of Birth</label>
                  </div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <input type="date" className="control-form" id="dateOfBirth" name="dateOfBirth" value={dateOfBirth} onChange={this.change} autoComplete="dateOfBirth" max="2000-01-01" pattern="[a-z|A-Z|\d]{1,15}" />
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                    <label className="white-text">Password</label>
                  </div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <input type="password" required="true" className="control-form" id="password" name="password" value={password} onChange={this.change} autoComplete="password" />
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                      <label className="white-text">Confirm Password</label>
                  </div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                      <input type="password" required="true" className="control-form" id="confirmPassword" name="confirmPassword" value={confirmPassword} onChange={this.change} autoComplete="confirmPassword" />
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                  <div className="col-2-6 col-6-6-md col-6-6-xs">
                    <button type="submit" className="submit-button button-white" id="dkkd" onClick={this.submitSignup}>Create Account</button>
                  </div>
                  <div className="col-2-6 col-6-6-md col-6-6-xs mt-4-md mt-4-xs">
                    <button type="submit" className="submit-button button-cancel" onClick={this.goToHome}>Cancel</button>
                  </div>
                </div>
                <div className="field grid-container">
                  <div className="col-2-6 col-6-6-md col-6-6-xs"></div>
                  <div className="col-4-6 col-6-6-md col-6-6-xs">
                    <span className="white-text">Already have an account?</span> <a href="signin.html" className="underline">Signin</a>
                  </div>
                </div>
              </form>
          </div>
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  success: state.user.success
});

Signup.propTypes = {
  SignupUser: PropTypes.func,
  success: PropTypes.bool
};

export default connect(mapStateToProps, {
  SignupUser
})(Signup);
