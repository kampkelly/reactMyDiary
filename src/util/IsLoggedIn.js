export const isLoggedIn = () => {
  if (localStorage.getItem('diary_token') !== null) {
    return true;
  }
  return false;
};
