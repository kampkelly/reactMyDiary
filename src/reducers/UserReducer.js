import { SIGNIN, SIGNUP, SIGNOUT, PROFILE, NOTIFICATION, UPDATE_PROFILE } from '../actionTypes/UserConstants';
import { asyncActionName } from '../util/AsyncUtil';


const initialState = {
  isAuth: false,
  loading: false,
  message: '',
  success: false,
  failure: false,
  user: {}
};

const UserReducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case asyncActionName(SIGNIN).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SIGNIN).success:
      return {
        ...state,
        success: true, ...action.payload.user, isAuth: true
      };
    case asyncActionName(SIGNIN).failure:
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(SIGNUP).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(SIGNUP).success:
      return {
        ...state,
        success: action.payload,
      };
    case asyncActionName(SIGNUP).failure:
      document.querySelector('.form_error_text').style.display = 'block';
      document.querySelector('.form_error_text small').textContent = action.payload.error;
      return {
        ...state,
      };
    case asyncActionName(PROFILE).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(PROFILE).success:
      return {
        ...state,
        success: true, user: action.payload
      };
    case asyncActionName(PROFILE).failure:
      return {
        ...state,
      };
    case asyncActionName(UPDATE_PROFILE).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(UPDATE_PROFILE).success:
      return {
        ...state,
        message: action.payload.message, success: true, user: action.payload.user
      };
    case asyncActionName(UPDATE_PROFILE).failure:
      document.querySelector('.form_error_text').style.display = 'block';
			document.querySelector('.form_error_text small').textContent = data.message;
      return {
        ...state,
      };
    case asyncActionName(NOTIFICATION).loading:
      return {
        ...state,
        loading: action.payload,
      };
    case asyncActionName(NOTIFICATION).success:
      return {
        ...state,
        success: true, message: action.payload.message, user: action.payload.user
      };
    case asyncActionName(NOTIFICATION).failure:
      return {
        ...state,
      };
    case asyncActionName(SIGNOUT).success:
      return initialState;
    default:
      return state;
  }
};

export default userReducer;
