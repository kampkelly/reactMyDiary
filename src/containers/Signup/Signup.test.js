import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Signup from './Signup';
import UserReducer from '../../reducers/UserReducer';
import { asyncActions } from '../../util/AsyncUtil';
import { SIGNUP } from '../../actionTypes/UserConstants';
import setAuthToken from '../../util/AuthTokenUtil';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  user: {
    user: {},
    message: '',
    success: false
  },
  email: 'email@example.com',
  fullName: 'full name',
  password: 'password',
  confirmPassword: 'password',
  entry: {
    entries: []
  }
});

const props = {
  history: {
    push: jest.fn()
  },
  submitSignup: jest.fn(),
  SignupUser: jest.fn(),
};

let component;
let myComponent;

describe('<Signup/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <Signup {...props} />
      </Provider>
    );
    document.body.innerHTML =
    '<main></main>';
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should show profile', () => {
    const payload = {};
    const expectedAction = {
      type: 'SIGNUP_SUCCESS',
      payload
    };
    expect(asyncActions(SIGNUP).success(payload)).toEqual(expectedAction);
  });
  it('should dispatch an action when signin is loading', () => {
    const payload = false;
    const action = {
      type: 'SIGNUP_LOADING',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({
      loading: payload
    });
  });
  it('should dispatch an action to signin a user', () => {
    const payload = true;
    const action = {
      type: 'SIGNUP_SUCCESS',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({
      success: action.payload,
    });
  });
  it('should dispatch an action when signin a user fails', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const payload = {};
    const action = {
      type: 'SIGNUP_FAILING',
      payload
    };
    const newState = UserReducer({}, action);
    expect(newState).toEqual({});
  });
  it('should signin user', () => {
    document.body.innerHTML =
    '<div>' +
    '  <span id="loading" />' +
    '  <span class="form_error_text" /><small></small><span>' +
    '  <button id="button" />' +
    '</div>';
    const emailInput = myComponent.find('input[name="email"]');
    const fullNameInput = myComponent.find('input[name="fullName"]');
    const passwordInput = myComponent.find('input[name="password"]');
    const confirmPasswordInput = myComponent.find('input[name="confirmPassword"]');
    emailInput.simulate('change', { target: { value: 'email@example.com' } });
    fullNameInput.simulate('change', { target: { value: 'full name' } });
    passwordInput.simulate('change', { target: { value: 'password' } });
    confirmPasswordInput.simulate('change', { target: { value: 'password' } });
    myComponent.find('button.button-white').simulate('click', {
      preventDefault: () => {
      }
     });
  });
  it('should click cancel button', () => {
    myComponent.find('button.button-cancel').simulate('click');
  });
  it('calls the signup user function', () => {
    setAuthToken('fffnf');
    setAuthToken('');
  });
});
