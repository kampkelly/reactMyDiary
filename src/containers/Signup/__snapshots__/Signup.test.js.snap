// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> should render without throwing an error 1`] = `
<Connect(Signup)
  SignupUser={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  submitSignup={[MockFunction]}
/>
`;
